src/components/electron/Filestream.js

```diff
@@ -19,13 +19,20 @@ class Filestream extends React.Component {
     super(props)
     this.state = {
       original: {},
-      intervalId: null
+      prevFileInstance: null
     }
   }
 
   componentDidUpdate(prevProps) {
     if (prevProps.file !== this.props.file) {
-      tail(this.props.file[0]).start(
+      const Filetail = new Tail(this.props.file[0])
+      // because we use a conditon we can:
+      // eslint-disable-next-line react/no-did-update-set-state
+      this.setState({ prevFileInstance: Filetail })
```

Comment:
Shouldn't this `setState` happen *after* the conditional `prevFileInstance.stop()` below it?

Reply:
No - the set.state saves the current file in state, so if on a later file change (user selected a new file) the interval can be stopped on this.state.prevFileInstance. The order doesn't matter - but because it's confusing I did put it after the conditional and renamed `prevFileInstance` to `currentFileInstance` in https://github.com/rowolff/babel-poe/commit/8d18bd30cf24aa5442c1e14cf46ec446c10b41b3